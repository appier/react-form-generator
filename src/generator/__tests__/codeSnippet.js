import {
  importComponent,
  embedFieldComponent,
  getValidateFnText
} from '../codeSnippet';

describe('codeSnippet Generator', () => {
  it('importComponent', () => {
    expect(
      importComponent({
        name: 'TextInputer',
        path: './Section1/TextInputer'
      })
    ).toMatchSnapshot();
  });

  it('embedFieldComponent', () => {
    expect(
      embedFieldComponent({
        componentName: 'TextInputer',
        key: 'name',
        changeHandler: 'update'
      })
    ).toMatchSnapshot();
  });

  it('getValidateFnText: default', () => {
    expect(getValidateFnText({ label: 'Name' })({})).toMatchSnapshot();
  });

  it('getValidateFnText: required', () => {
    expect(
      getValidateFnText({ label: 'Name' })({ required: true })
    ).toMatchSnapshot();
  });

  it('getValidateFnText: required, minLength', () => {
    expect(
      getValidateFnText({ label: 'Name' })({
        required: true,
        minLength: 3
      })
    ).toMatchSnapshot();
  });

  it('getValidateFnText: required, minLength, maxLength', () => {
    expect(
      getValidateFnText({ label: 'Name' })({
        required: true,
        minLength: 3,
        maxLength: 20
      })
    ).toMatchSnapshot();
  });

  it('getValidateFnText: minLength, maxLength', () => {
    expect(
      getValidateFnText({ label: 'CompanyName' })({
        minLength: 3,
        maxLength: 20
      })
    ).toMatchSnapshot();
  });

  it('getValidateFnText: number', () => {
    expect(
      getValidateFnText({ label: 'Bid Price' })({
        number: true
      })
    ).toMatchSnapshot();
  });

  it('getValidateFnText: require, number', () => {
    expect(
      getValidateFnText({ label: 'Bid Price' })({
        required: true,
        number: true
      })
    ).toMatchSnapshot();
  });

  it('getValidateFnText: number, min, max', () => {
    expect(
      getValidateFnText({ label: 'Bid Price' })({
        number: true,
        min: 1,
        max: 15
      })
    ).toMatchSnapshot();
  });
});
