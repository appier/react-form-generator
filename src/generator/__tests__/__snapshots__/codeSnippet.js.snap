exports[`codeSnippet Generator embedFieldComponent 1`] = `
"<TextInputer
  value={name}
  registerValidator={undefined}
  onChange={update}
/>"
`;

exports[`codeSnippet Generator getValidateFnText: default 1`] = `"return null;"`;

exports[`codeSnippet Generator getValidateFnText: minLength, maxLength 1`] = `
"
if(value.length < 3){
  return \'CompanyName must be longer than 3 chars\';
}
else if(value.length > 20){
  return \'CompanyName must be smaller than 20 chars\';
}
else {
  return null;
}
"
`;

exports[`codeSnippet Generator getValidateFnText: number 1`] = `
"
if(isNaN(Number(value))){
  return \'Bid Price must be a valid Number\';
}
else {
  return null;
}
"
`;

exports[`codeSnippet Generator getValidateFnText: number, min, max 1`] = `
"
if(isNaN(Number(value))){
  return \'Bid Price must be a valid Number\';
}
else if(Number(value) < 1){
  return \'Bid Price must be larger than 1\';
}
else if(Number(value) > 15){
  return \'Bid Price must be smaller than 15\';
}
else {
  return null;
}
"
`;

exports[`codeSnippet Generator getValidateFnText: require, number 1`] = `
"
if(value == null || value === \'\'){
  return \'Bid Price cannot be empty\';
}
else if(isNaN(Number(value))){
  return \'Bid Price must be a valid Number\';
}
else {
  return null;
}
"
`;

exports[`codeSnippet Generator getValidateFnText: required 1`] = `
"
if(value == null || value === \'\'){
  return \'Name cannot be empty\';
}
else {
  return null;
}
"
`;

exports[`codeSnippet Generator getValidateFnText: required, minLength 1`] = `
"
if(value == null || value === \'\'){
  return \'Name cannot be empty\';
}
else if(value.length < 3){
  return \'Name must be longer than 3 chars\';
}
else {
  return null;
}
"
`;

exports[`codeSnippet Generator getValidateFnText: required, minLength, maxLength 1`] = `
"
if(value == null || value === \'\'){
  return \'Name cannot be empty\';
}
else if(value.length < 3){
  return \'Name must be longer than 3 chars\';
}
else if(value.length > 20){
  return \'Name must be smaller than 20 chars\';
}
else {
  return null;
}
"
`;

exports[`codeSnippet Generator importComponent 1`] = `"import TextInputer from \'./Section1/TextInputer\';"`;
